browser/themes/brave_theme_utils.cc

```diff
@@ -15,28 +16,15 @@ namespace ui {
 // static
 void BraveThemeUtils::SetDarkMode(bool dark_mode) {
   NativeTheme::GetInstanceForNativeUi()->set_dark_mode(dark_mode);
+  NativeTheme::GetInstanceForWeb()->set_dark_mode(dark_mode);
 }
 
 // static
-void BraveThemeUtils::SetPreferredColorScheme(BraveThemeType brave_theme_type) {
```

Comment:
For re-calculating color scheme, I think we just need to set dark mode in native theme before calling `CalculatePreferredColorScheme`.